crates/swc_css_minifier/src/compressor/calc_sum.rs

```diff
@@ -0,0 +1,1102 @@
+                        &mut data.value.value,
+                        prev_operator,
+                    );
+                    CalcSumContext::update_calc_value(
+                        &mut self.expressions,
+                        *pos,
+                        CalcValueOrOperator::Value(CalcValue::Percentage(data.clone())),
+                    );
+                } else {
+                    self.push(operator, operand);
+                }
+            }
+            None => self.percentage = Some(self.new_indexed_data(operator, operand, p.clone())),
+        }
+    }
+
+    fn sum_length(&mut self, operator: &Option<CalcOperator>, operand: &CalcProduct, l: &Length) {
+        let unit = l.unit.value.to_ascii_lowercase();
+        if is_absolute_length(&unit) {
+            self.sum_absolute_length(operator, operand, l)
+        } else {
+            self.sum_other_length(operator, operand, l)
+        }
+    }
+
+    fn sum_absolute_length(
+        &mut self,
+        operator: &Option<CalcOperator>,
+        operand: &CalcProduct,
+        l: &Length,
+    ) {
+        match &mut self.absolute_length {
+            Some(IndexedOperatorAndOperand {
+                operator: prev_operator,
+                operand: IndexedData { pos, data },
+            }) => {
+                let prev_unit = data.unit.value.to_ascii_lowercase();
+                let unit = l.unit.value.to_ascii_lowercase();
+                if let Some(result) =
+                    get_absolute_length_ratio(&prev_unit, &unit).and_then(|ratio| {
+                        CalcSumContext::try_to_sum_values(
+                            prev_operator,
+                            operator,
+                            data.value.value,
+                            l.value.value,
+                            Some(*ratio),
+                        )
+                    })
+                {
+                    data.value.value = result;
+
+                    CalcSumContext::switch_sign_if_needed(
+                        &mut self.expressions,
+                        &mut data.value.value,
+                        prev_operator,
+                    );
+                    CalcSumContext::update_calc_value(
+                        &mut self.expressions,
+                        *pos,
+                        CalcValueOrOperator::Value(CalcValue::Dimension(Dimension::Length(
+                            data.clone(),
+                        ))),
+                    );
+                } else {
+                    self.push(operator, operand);
+                }
+            }
+            None => {
+                self.absolute_length = Some(self.new_indexed_data(operator, operand, l.clone()))
+            }
+        }
+    }
+
+    fn sum_other_length(
+        &mut self,
+        operator: &Option<CalcOperator>,
+        operand: &CalcProduct,
+        l: &Length,
+    ) {
+        let unit = l.unit.value.to_ascii_lowercase().to_string();
```

Comment:
This is inefficient. Please remove `to_string()`

Reply:
You can change `other_lengths` to `AHashMap<JsWord, IndexedOperatorAndOperand<Length>>`